65 page forum

PyotrGrogorchenko opened the pull request.

PyotrGrogorchenko committed:
[65_PageForum] Forum ipi connect

PyotrGrogorchenko committed:
[65_PageForum] Forum api connect

PyotrGrogorchenko committed:
[65_PageForum] Page forum api connection

PyotrGrogorchenko committed:
[65_PageForum] Autentification, create topic

PyotrGrogorchenko committed:
[65_PageForum] Create/update topic

PyotrGrogorchenko committed:
[65_PageForum] Topic list, topic reducer

PyotrGrogorchenko committed:
[65_PageForum] Topic, ListTopics

PyotrGrogorchenko committed:
[65_PageForum] Forum

PyotrGrogorchenko committed:
[65_PageForum] Forum

realb0t commented on src/components/forum/Topic/Topic.tsx:
```diff
@@ -1,65 +1,164 @@
+        classes, cb, id, isNew
+    } = props
+
+    const [state, dispatch] = useReducer(reducer, _.cloneDeep(initialState))
```
Вроде Саша делал какой-то deep clone. Посмотри на него, потому что тащить lodash для такой задачи - такое. И кстати lodash может негативно влиять на бандл в таком импорте. Возможно, если он всетаки понадобится, его нужно импортнуть так `'lodash/cloneDeep'`

realb0t commented on src/components/forum/Topic/Topic.tsx:
```diff
@@ -1,65 +1,164 @@
-        if (cb) cb()
+        dispatch(reset())
+        window.alertHide()
+        cb()
```
а тут не должно быть зависимости на `cb`?

realb0t commented on src/components/forum/Topic/Topic.tsx:
```diff
@@ -1,65 +1,164 @@
+    const onClickDelete = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        e.preventDefault()
+        deleteTopic(fields, userInfo)
+            .then((sucess) => {
```
тут немного не понятен кейс когда `!!sucсess !== true` и почему выполняется этот каллбек?

realb0t commented on src/components/forum/Topic/utils/fillTopic.ts:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    return res.data
+    // const {record, readOnly} = res.data
```
комментарии убираем )

realb0t commented on src/components/forum/commentsTree/Comment/Comment.tsx:
```diff
@@ -18,7 +21,8 @@ const Comment: FC<Props> = (props: Props) => {
         e.preventDefault()
-        setShow(false)
+        saveComment(userInfo)
+        // setShow(false)
```
снова комментарии убираем

realb0t commented on src/components/forum/commentsTree/Comment/utils/saveComment.ts:
```diff
@@ -0,0 +1,7 @@
+export const saveComment = async (userInfo: UserInfoEmpty) => {
+    if (!userInfo) return
+    // eslint-disable-next-line no-console
+    console.log('saveComment')
```
дебаг тоже убираем

realb0t commented on src/pages/Forum/Forum.tsx:
```diff
@@ -1,53 +1,72 @@
+    useEffect(() => {
+        logIn(userInfo)
+            .then((res) => {
+                if (res) dispatch(setAvailable())
```
снова fulfilled вызывается с каким-то флагом. старайся такого избегать

realb0t commented on src/pages/Forum/reducer/reducer.ts:
```diff
@@ -0,0 +1,43 @@
+import {State, Action} from './types'
+
+export const reducer = (state: State, action: Action): State => {
+    // const {payload} = action
```
комметарии удаляем )

realb0t commented on src/pages/Forum/utils/logIn.ts:
```diff
@@ -0,0 +1,11 @@
+import {UserInfoEmpty} from '../../../store/reducers/user/state'
+import {postLogIn} from '../../../services/API/db/index'
+
+export const logIn = async (userInfo: UserInfoEmpty) => {
```
Наверное эти флаги в fulfilled каллбеках тебе нужны, потому что ты не определяешь возвращаемое значение типа `Promise<Result>` и у тебя `Result` может быть `undefined`?

PyotrGrogorchenko committed:
[65_PageForum] Comments

PyotrGrogorchenko committed:
[65_PageForum] Comments

PyotrGrogorchenko committed:
[65_PageForum] Merged main

PyotrGrogorchenko committed:
[65_PageForum] Removed colors package from api

PyotrGrogorchenko committed:
[65_PageForum] Reactions

365g1s22 committed:
game#65 Fix docker build locally

1) fix spa
2) change postgres hostname, port
3) fix problem with validation
4) add .dockerignore

365g1s22 committed:
Merge branch '65_PageForum' into feature/game#65-forum

365g1s22 committed:
game#65 Run prod and dev without changes

365g1s22 committed:
game#65 Fix dockerfile

365g1s22 committed:
game#65 Remove layout index

PyotrGrogorchenko committed:
[65_PageForum] Emotions

365g1s22 committed:
game#65 Fix strange name error

365g1s22 committed:
game#65 Change cors to *

365g1s22 committed:
Merge branch '65_PageForum' into feature/game#65-forum
